src/app/qgsprojectproperties.cpp

```diff
@@ -1907,60 +1908,28 @@ void QgsProjectProperties::pbnWCSLayersDeselectAll_clicked()
 
 void QgsProjectProperties::pbnLaunchOWSChecker_clicked()
 {
-  QString myStyle = QgsApplication::reportStyleSheet();
-  teOWSChecker->clear();
-  teOWSChecker->document()->setDefaultStyleSheet( myStyle );
-  teOWSChecker->setHtml( "<h1>" + tr( "Start checking QGIS Server" ) + "</h1>" );
-
-  QStringList owsNames, encodingMessages;
-  checkOWS( QgisApp::instance()->layerTreeView()->layerTreeModel()->rootGroup(), owsNames, encodingMessages );
-
-  QStringList duplicateNames, regExpMessages;
-  QRegExp snRegExp = QgsApplication::shortNameRegExp();
-  const auto constOwsNames = owsNames;
-  for ( const QString &name : constOwsNames )
-  {
-    if ( !snRegExp.exactMatch( name ) )
-      regExpMessages << tr( "Use short name for \"%1\"" ).arg( name );
-    if ( duplicateNames.contains( name ) )
-      continue;
-    if ( owsNames.count( name ) > 1 )
-      duplicateNames << name;
-  }
+  QList<QgsProjectServerValidator::ValidationResult> validationResults;
+  bool results = QgsProjectServerValidator::validate( QgsProject::instance(), validationResults );
 
-  if ( !duplicateNames.empty() )
-  {
-    QString nameMessage = "<h1>" + tr( "Some layers and groups have the same name or short name" ) + "</h1>";
-    nameMessage += "<h2>" + tr( "Duplicate names:" ) + "</h2>";
-    nameMessage += duplicateNames.join( QStringLiteral( "</li><li>" ) ) + "</li></ul>";
-    teOWSChecker->setHtml( teOWSChecker->toHtml() + nameMessage );
-  }
-  else
+  QString errors;
+  if ( !results )
   {
-    teOWSChecker->setHtml( teOWSChecker->toHtml() + "<h1>" + tr( "All names and short names of layer and group are unique" ) + "</h1>" );
-  }
-
-  if ( !regExpMessages.empty() )
-  {
-    QString encodingMessage = "<h1>" + tr( "Some layer short names have to be updated:" ) + "</h1><ul><li>" + regExpMessages.join( QStringLiteral( "</li><li>" ) ) + "</li></ul>";
-    teOWSChecker->setHtml( teOWSChecker->toHtml() + encodingMessage );
-  }
-  else
-  {
-    teOWSChecker->setHtml( teOWSChecker->toHtml() + "<h1>" + tr( "All layer short names are well formed" ) + "</h1>" );
-  }
-
-  if ( !encodingMessages.empty() )
-  {
-    QString encodingMessage = "<h1>" + tr( "Some layer encodings are not set:" ) + "</h1><ul><li>" + encodingMessages.join( QStringLiteral( "</li><li>" ) ) + "</li></ul>";
-    teOWSChecker->setHtml( teOWSChecker->toHtml() + encodingMessage );
+    for ( const QgsProjectServerValidator::ValidationResult &result : qgis::as_const( validationResults ) )
+    {
+      errors += QLatin1String( "<b>" ) % QgsProjectServerValidator::displayValidationError( result.error ) % QLatin1String( " :</b> " );
+      errors += result.identifier.toString();
+    }
   }
   else
   {
-    teOWSChecker->setHtml( teOWSChecker->toHtml() + "<h1>" + tr( "All layer encodings are set" ) + "</h1>" );
+    errors += QString( tr( "Ok, it seems valid." ) );
```

Comment:
```suggestion
    errors += tr( "Ok, it seems valid." );
```

Reply:
Or more professionally: "Project is valid" / "Project validated successfully"